### PR TITLE
Hw 18.1 authentication

### PR DESCRIPTION
**Pull request by Valdes2**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/controller/AddItemToBucket.java [hunk 1]
```diff
@@ -24,8 +23,9 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
+        Long userId =(Long) req.getSession(true).getAttribute("userId");
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/controller/AddItemToBucket.java]:
```suggestion
        Long userId = Long.valueOf(req.getSession(true).getAttribute("userId"));
```

File: src/main/java/mate/academy/internetshop/controller/AddUserContrroller.java [hunk 2]
```diff
@@ -3,17 +3,15 @@
 import java.io.IOException;
 
 import javax.servlet.ServletException;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.*;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/controller/AddUserContrroller.java]: remove *

File: src/main/java/mate/academy/internetshop/controller/BucketController.java [hunk 3]
```diff
@@ -11,18 +11,23 @@
...
     @Inject
     private static BucketService bucketService;
+    @Inject
+    private static UserService userService;
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        Long bucketId = Storage.buckets.get(BUCKET_INDEX).getId();
+       // Long bucketId = Storage.buckets.get(BUCKET_INDEX).getId();
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/controller/BucketController.java]: remove commented code from repository

File: src/main/java/mate/academy/internetshop/controller/LoginController.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package mate.academy.internetshop.controller;
+
+import mate.academy.internetshop.exceptions.AuthenticationException;
+import mate.academy.internetshop.lib.Inject;
+import mate.academy.internetshop.model.User;
+import mate.academy.internetshop.service.UserService;
+import org.apache.log4j.Logger;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.*;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/controller/LoginController.java]: remove *

File: src/main/java/mate/academy/internetshop/controller/OrdersListController.java [hunk 5]
```diff
@@ -27,7 +26,7 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        Long userId = Storage.buckets.get(BUCKET_INDEX).getUserId();
+        Long userId =(Long) req.getSession(true).getAttribute("userId");
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/controller/OrdersListController.java]: the same)

File: src/main/java/mate/academy/internetshop/web/MainServlet.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package mate.academy.internetshop.web;
+
+public class MainServlet {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/web/MainServlet.java]: do we need this empty class?

File: src/main/webapp/WEB-INF/views/addUser.jsp [hunk 7]
```diff
@@ -9,24 +9,30 @@
...
             <button class="w3-btn w3-blue-grey" type="submit">Enter</button>
         </form>
     </div>
-    <div class="w3-container">
-        <form action="${pageContext.request.contextPath}/index">
-            <input class="w3-btn w3-dark-gray" type="submit" value="Return to main page" />
-        </form>
-    </div>
 </body>
 </html>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/addUser.jsp]: add a blank line at the end of the file

File: src/main/webapp/WEB-INF/views/getAllUsers.jsp [hunk 8]
```diff
@@ -4,41 +4,49 @@
...
+                    </td>
+                    <td>
+                        <a class="w3-btn w3-greyscale" href="${pageContext.request.contextPath}/servlet/deleteUser?user_id=${user.id}">Delete</a>
+                    </td>
+                </tr>
+            </c:forEach>
+        </table>
+    </div>
 </body>
 </html>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/getAllUsers.jsp]: :)

**Review by Sofasmile — COMMENTED** [review]
good, but make small fixes

File: src/main/java/mate/academy/internetshop/controller/AddItemToBucket.java (hunk 1 above)
Comment (Valdes2) [reply src/main/java/mate/academy/internetshop/controller/AddItemToBucket.java]: fixed

File: src/main/java/mate/academy/internetshop/controller/AddUserContrroller.java (hunk 2 above)
Comment (Valdes2) [reply src/main/java/mate/academy/internetshop/controller/AddUserContrroller.java]: fixed

File: src/main/java/mate/academy/internetshop/controller/BucketController.java (hunk 3 above)
Comment (Valdes2) [reply src/main/java/mate/academy/internetshop/controller/BucketController.java]: fixed

File: src/main/java/mate/academy/internetshop/controller/LoginController.java (hunk 4 above)
Comment (Valdes2) [reply src/main/java/mate/academy/internetshop/controller/LoginController.java]: fixed

File: src/main/java/mate/academy/internetshop/controller/OrdersListController.java (hunk 5 above)
Comment (Valdes2) [reply src/main/java/mate/academy/internetshop/controller/OrdersListController.java]: fixed

File: src/main/java/mate/academy/internetshop/web/MainServlet.java (hunk 6 above)
Comment (Valdes2) [reply src/main/java/mate/academy/internetshop/web/MainServlet.java]: fixed

File: src/main/webapp/WEB-INF/views/addUser.jsp (hunk 7 above)
Comment (Valdes2) [reply src/main/webapp/WEB-INF/views/addUser.jsp]: fixed

File: src/main/webapp/WEB-INF/views/getAllUsers.jsp (hunk 8 above)
Comment (Valdes2) [reply src/main/webapp/WEB-INF/views/getAllUsers.jsp]: fixed)

File: .gitignore [hunk 9]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
 .target/*
+.src/main/resources/log/*
```
Comment (Sofasmile) [inline .gitignore]: add a blank line at the end of the file

**Review by Sofasmile — APPROVED** [review]

File: .gitignore (hunk 9 above)
Comment (Valdes2) [reply .gitignore]: fixed

**Pull request merged**